components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/DefaultOAuth2ScopeValidator.java

```diff
@@ -177,8 +178,11 @@ private List<String> getAuthorizedScopes(List<String> requestedScopes, Authentic
         if (requestedScopes.isEmpty()) {
             return approvedScopes;
         }
-        List<String> internalScopes = getInternalScopes(tenantDomain);
+        List<String> internalScopes = new ArrayList<>();
         if (requestedScopes.contains(SYSTEM_SCOPE)) {
+            if (CollectionUtils.isEmpty(internalScopes)) {
```

Comment:
do we need the empty check here?

Reply:
added this as an additional check if we resolve this in outer block when doing improvements so that performance will be not impacted